src/Tribe/Main.php

```diff
@@ -1488,18 +1506,72 @@ public function issue_noindex() {
 				return;
 			}
 
-			// By default, we add a noindex tag for all month view requests and any other
-			// event views that are devoid of events
-			$add_noindex  = ( ! $wp_query->have_posts() || Month_View::get_view_slug() === $context->get( 'view' ) );
+			$view = $context->get( 'view' );
+
+			$start_date = ! empty( $wp_query->query[ 'eventDate' ] ) ? $wp_query->get( 'eventDate' ) : $context->get( 'now' );
+			$start_date = Tribe__Date_Utils::build_date_object( $start_date );
+
+			/**
+			 * Allow specific views to hook in and add their own calculated events.
+			 * This *bypasses* the cached query immediately after it.
+			 *
+			 * @since TBD
+			 *
+			 * @param ?Tribe__Repository|null $events     The events repository. False if not hooked in to.
+			 * @param DateTime                $start_date The start date (object) of the query.
+			 * @param Tribe__Context          $context    The current context.
+			 *
+			 */
+			$events = apply_filters( 'tec_events_noindex', null, $start_date, $context );
+
+			$foo = null;
+
+			// If nothing has hooked in ($events is boolean false), we assume a list-style view (no end-date limiter)
+			//  with no params and do a quick query for a single event after the start date.
+			if ( null === $events ) {
+				$cache     = new Tribe__Cache();
+				$trigger   = Tribe__Cache_Listener::TRIGGER_SAVE_POST;
+				$cache_key = $cache->make_key(
+					[
+						'context' => $context,
```

Comment:
The `Tribe__Cache::make_key` method will `serialize` what you pass it.
The `Tribe__Context` object is a large one; by serializing it you're getting a very large string that has to moved back and forth. Furthermore, the serialization of the context will eagerly resolve **all its locations**.
Pick the parts of the context you care about for this, build your own hashed key (reduce to a fixed length) and use to store the value.